src/mscorlib/src/System/String.cs

```diff
@@ -2559,7 +2559,11 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
                     }
                     return (value.Length == 1) ?
                             true :                 // First char is the same and thats all there is to compare
+#if FEATURE_CORECLR
```

Comment:
The FEATURE_CORECLR ifdefs are not necessary for performance improvements like this one.